### ISSUE TITLE
Can't enable basic types like those found in C2

### ISSUE THREAD

**Issue by jdknox** [c3lang/c3c#660]
How can we enable basic type aliases like those found in C2: http://www.c2lang.org/site/language/basic_types/

For example: `u32` `u16` etc.

Alternatively, there should be a way to disable typography restrictions on _basic_ types as that isn't related to programming.

Just to be clear: `uint` doesn't start with a capital letter, in the exact same way that `u32` doesn't.  So the typography restriction already has a precedent for being ignored on _basic_ types.  This is inconsistent, right?

**Comment by data-man** [comment]
https://github.com/c3lang/c3c/discussions/434

**Comment by lerno** [comment]
There is no way to have a type start with a lower case character unless it is built into the compiler, that is, the type is part of the language spec. Originally I provided both int and i32, uint and u32 etc. However, the motivation for it was very weak: "add some naming flexibility for the ones that prefer a different basic type naming standard". Using a similar argument one would then provide int32 and uint32. After all, some prefer that etc.

So I ended up removing those aliases from the language, picking int over i32 (motivation can be found in #434 – basically readability issues). Furthermore even for types that have bitsizes (I use it for int128 and float128) I ended up preferring the longer names for their clarity in those new types.

The rule then for built in types is that either they are a fundamental known size type, or they are one of the main platform dependent types, such as `usz`.
